### PR TITLE
UI tests/aag 3058 interstitial UI tests

### PR DESCRIPTION
**Pull request by thomas-james-1986**
## JIRA
[[AAG-3058](https://superawesomeltd.atlassian.net/browse/AAG-3058)]

## DESCRIPTION

This PR adds UI tests for the functionality of interstitials:

1) Rendering the ad content
2) The close button configurations
3) The safe ad visibility configurations
4) The bumper page behaviour
5) The parental gate behaviour

## SCREENSHOTS

Simulator Screen Shot - iPhone 14 Pro - 2023-04-20 at 01 44 30

[AAG-3058]: https://superawesomeltd.atlassian.net/browse/AAG-3058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Example/SuperAwesomeExampleUITests/Stories/InterstitialUITests.swift [hunk 1]
```diff
@@ -7,58 +7,208 @@
...
+            $0.tapPlacement(withName: "Mobile Interstitial Flat Colour Portrait")
+
+            interstitialScreen(app) { interstitialScreen in
+                interstitialScreen.waitForView()
+                // Check that the close button is not visible at first
+                interstitialScreen.checkCloseButtonDoesNotExist()
+                // Wait for the close button to be visible
+                interstitialScreen.waitForCloseButton()
+            }
+        }
```
Comment (me2k) [inline Example/SuperAwesomeExampleUITests/Stories/InterstitialUITests.swift:212]: Can you add a test that combines the parent gate and the bumper please

**Review by me2k — CHANGES_REQUESTED** [review]
Awesome work Tom, really pleased to see this! 
Can you add a test that combines the parent gate and the bumper please.

File: Example/SuperAwesomeExampleUITests/Stories/InterstitialUITests.swift (hunk 1 above)
Comment (thomas-james-1986) [reply Example/SuperAwesomeExampleUITests/Stories/InterstitialUITests.swift:251]: Added!

**Review by gunhansancar — DISMISSED** [review]

File: Example/SuperAwesomeExampleUITests/Robots/ParentGateRobot.swift [hunk 2]
```diff
@@ -8,50 +8,85 @@
...
+            elements.count == 2,
+            let firstNumber = Int.parse(from: String(elements[0])),
+            let secondNumber = Int.parse(from: String(elements[1]))
+        else {
+            XCTFail("Failed to extract numbers for the parental gate question")
+            return ""
+        }
+                
+        return String(firstNumber + secondNumber)
+    }
```
Comment (me2k) [inline Example/SuperAwesomeExampleUITests/Robots/ParentGateRobot.swift:79]: Genius! I was going to mock, but this is way cooler

**Review by me2k — APPROVED** [review]
Awesome job Tom, thank you for adding that test.

**Review by me2k — APPROVED** [review]

**Pull request merged**